### PR TITLE
ukify: Optimize initrd joining

### PR DESCRIPTION
**Pull request by daandemeyer**
Let's not load all the initrds into memory, but use a temporary file instead.

### CODE REVIEW

**Comment by keszybz** [comment]
So… you said "optimize". Is it faster?

**Comment by keszybz** [comment]
(The code looks good, but it's also quite a bit of complexity. If we do this, I think we should have some concrete numbers to back up the decision.)

File: src/ukify/ukify.py [hunk 1]
```diff
@@ -425,19 +430,33 @@ def call_systemd_measure(uki, linux, opts):
...
+            with open(i, "rb") as f:
+                sz = os.stat(f.fileno()).st_size
 
-    seq = []
-    for file in initrds:
-        initrd = file.read_bytes()
-        padding = b'\0' * round_up(len(initrd), 4)  # pad to 32 bit alignment
-        seq += [initrd, padding]
+                try:
+                    fcntl.fcntl(tmp.fileno(), FICLONERANGE, struct.pack("qQQQ", f.fileno(), 0, sz, off))
```
Comment (poettering) [inline src/ukify/ukify.py:450]: hmm, isn't the api do jour to use copy_file_range() these days?

python even exposes that directly as os.copy_file_range()

**Comment by DaanDeMeyer** [comment]
This was a red herring and not actually what was making things slow, closing

**Pull request closed without merging**